bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+          the same transaction, in the same block, but for a
+          different `node_id_1` or `node_id_2`:
+            - SHOULD blacklist the previous message's `node_id_1`
+              and `node_id_2`, as well as this `node_id_1` and `node_id_2` AND
+              forget any channels connected to them.
+        - otherwise:
+            - SHOULD store this `channel_announcement_2`.
+- once its funding output has been spent OR reorganized out:
+    - SHOULD forget a channel after a 12-block delay.
+
+### Rationale
+
+Both nodes are required to sign to indicate they are willing to route other
+payments via this channel (i.e. be part of the public network); requiring their
+aggregated MuSig2 Schnorr signature proves that they control the channel.
+
+The blacklisting of conflicting nodes disallows multiple different
+announcements. Such conflicting announcements should never be broadcast by any
+node, as this implies that keys have leaked.
+
+While channels should not be advertised before they are sufficiently deep, the
+requirement against rebroadcasting only applies if the transaction has not moved
+to a different block.
+
+In order to avoid storing excessively large messages, yet still allow for
+reasonable future expansion, nodes are permitted to restrict rebroadcasting
+(perhaps statistically).
+
+New channel features are possible in the future: backwards compatible (or
+optional) features will have _odd_ feature bits, while incompatible features
+will have _even_ feature bits
+(["It's OK to be odd!"](00-introduction.md#glossary-and-terminology-guide)).
+
+A delay of 12-blocks is used when forgetting a channel on funding output spend
+as to permit a new `channel_announcement_2` to propagate which indicates this
+channel was spliced.
+
+## The `node_announcement_2` Message
+
+This gossip message, like the legacy `node_announcement` message, allows a node 
+to indicate extra data associated with it, in addition to its public key. 
+To avoid trivial denial of service attacks, nodes not associated with an already
+known channel (legacy or taproot) are ignored.
+
+Unlike the legacy `node_announcement` message, this message makes use of a 
+Schnorr signature instead of an ECDSA one. This will allow nodes to be backed 
+by multiple keys since MuSig2 can be used to construct the single signature.
+
+The other two main differences from the legacy message are that the timestamp 
+is now a block height and that the message is purely TLV based.
+
+1. type: xxx (`node_announcement_2`)
+2. data: (tlv_stream):
+    1. type: 0 (`announcement_sig`)
+    2. data:
+        * [`bip340_sig`:`bip340_sig`]
+    3. type: 1 (`features`)
+    4. data:
+        * [`...*byte`: `features`]
+    5. type: 2 (`timestamp`)
+    6. data:
+        * [`u32`: `block_height`]
+    7. type: 3 (`node_ID`)
+    8. data: 
+        * [`point`:`node_id`]
+    9. type: 4 (`color`)
+    10. data:
+         * [`rgb_color`:`rgb_color`]
```

Comment:
Maybe remove the color field? Don't think it was ever seriously used....

On the other hand, if we go TLV everywhere, then it matters less as including it is optional.